### ISSUE TITLE
bindsym shift+XF86BrightnessUp/Down not working

### ISSUE THREAD

**Issue by itaton** [swaywm/sway#3540]
* Sway Version: 1.0-beta.2-238-ge134da30

* Debug Log: Not necessary in my opinion, but if you need it, let me know

* Configuration File: https://gist.github.com/itaton/3dcac91f388549fa9ddf0e18171c6b73

`bindsym Shift+XF86MonBrightnessUp` does not work. It triggers the command defined in `bindsym XF86MonBrightnessUp` (without shift)  
The same issue arises with `XF86MonBrightnessDown`
The relevant lines in the config file are 206-218.  
Bindings with the shift key for other XF86 keys work, eg. XF86AudioRaiseVolume.

**Comment by Steverman** [comment]
I also noticed that `FN+XF86MonBrightness{Up,Down}` triggers two events with `xev -event keyboard`:
```log
MappingNotify event, serial 32, synthetic NO, window 0x0,
    request MappingKeyboard, first_keycode 8, count 248

MappingNotify event, serial 32, synthetic NO, window 0x0,
    request MappingKeyboard, first_keycode 8, count 248
```

`FN+Print` does not generate anything at all but still functions correctly. Using other modifiers have the expected event output. Also applies to `XF86MonBrightness{Up,Down}`.

Not sure if this helps.

**Comment by emersion** [comment]
Can you post a trace of `Shift+XF86MonBrightnessUp` in `xev`?

**Comment by Steverman** [comment]
MappingNotify event occurs only when `XF86MonBrightnessUp` is pressed. On keypress:
```
KeyPress event, serial 25, synthetic NO, window 0xe00001,
    root 0x39c, subw 0x0, time 365906, (345,533), root:(1305,562),
    state 0x0, keycode 50 (keysym 0xffe1, Shift_L), same_screen YES,
    XLookupString gives 0 bytes:
    XmbLookupString gives 0 bytes:
    XFilterEvent returns: False

MappingNotify event, serial 28, synthetic NO, window 0x0,
    request MappingKeyboard, first_keycode 8, count 248

MappingNotify event, serial 28, synthetic NO, window 0x0,
    request MappingKeyboard, first_keycode 8, count 248
```
And when I release it:
```
MappingNotify event, serial 30, synthetic NO, window 0x0,
    request MappingKeyboard, first_keycode 8, count 248

MappingNotify event, serial 30, synthetic NO, window 0x0,
    request MappingKeyboard, first_keycode 8, count 248

KeyRelease event, serial 30, synthetic NO, window 0xe00001,
    root 0x39c, subw 0x0, time 367715, (345,533), root:(1305,562),
    state 0x1, keycode 50 (keysym 0xffe1, Shift_L), same_screen YES,
    XLookupString gives 0 bytes:
    XFilterEvent returns: False
```
`MappingNotify event` always occur 4 times. On keypress and on release. Holding the modifiers down and repeatedly pressing `XF86MonBrightnessUp` won't generate additional events.

**Comment by jakubn551** [comment]
i have same isue but i fixed like this: install utility called evtest and xinput, via evtest found on what event andress is XF86Brightness generating output and add to /etc/X11/xorg.conf.d/ new file with this
 Section "InputClass"
    Identifier "event"
    Driver "libinput"
    Option "Device"   "dev/input/event*"
EndSection
then i restarted and keys started to work

**Comment by emersion** [comment]
X11 configuration isn't going to affect sway.

**Comment by schauveau** [comment]
I see the same problem on my laptop. I suspect that the cause is that the FN+brightness keys are  handled by a dedicated input device.  
For instance, on my laptop, the brightness keys are handled by /dev/input/event8 "Video Bus"  while the standard keyboard is /dev/input/event0 "AT Translated Set 2 keyboard".
You can test your input devices individually with `sudo evtest` 

The mapping notify events are probably caused by Sway switching between the 2 inputs.

**Comment by RedSoxFan** [comment]
I'm going to close this as a duplicate of #2897 since this one is newer. However, schauveau's comment appears to be correct. There is likely an issue with sharing modifiers across the keyboards in a seat.

**Comment by pat-s** [comment]
@schauveau 	This holds also true for me. The backlight keys are on "Video Bus". How can I add them so they are recognized? Using `swaymsg -t get_inputs` shows only touchpad and keyboard.

**Comment by schauveau** [comment]
If you miss some input devices then you may be suffering from https://github.com/swaywm/sway/issues/3843 (see also https://github.com/CanonicalLtd/lightdm/issues/63 ).
Are you starting sway from the linux console or from a display manager?

**Comment by pat-s** [comment]
Ah thanks, it is exactly that issue. Starting `sway` from tty works. I also used Lightdm.

**Comment by marcosnils** [comment]
I know this is closed but I'm using GDM 42.0 and I have a similar issue. I can see the keyboard inputs with `swaymsg -t get_inputs` but I can't seem to use Shift+XF86MonBrightnessUp. Binding XF86MonBrightnessUp alone actually works. 

`wev` or `xev` doesn't show any outputs when pressing either the special (fn+f6 in my keyboard) key.

Any clues?

**Comment by marcosnils** [comment]
nvm.. that wasn't the issue, following this commented fixed it: https://github.com/swaywm/sway/issues/6187#issuecomment-820756588
